src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -1,45 +1,58 @@
 package seedu.address.logic.commands;
 
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.Messages;
 import seedu.address.commons.core.index.Index;
 import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.util.Category;
 import seedu.address.model.Model;
 import seedu.address.model.account.ActiveAccount;
+import seedu.address.model.account.entry.Expense;
+import seedu.address.model.account.entry.Revenue;
+
+import static java.util.Objects.requireNonNull;
+
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Deletes an entry identified using it's displayed index from the revenue/expense list.
  */
 public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
+            + ": Deletes the Entry identified by the index number used in the displayed entry list.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Entry: %1$s";
 
     private final Index targetIndex;
+    private final Category category;
 
-    public DeleteCommand(Index targetIndex) {
+    public DeleteCommand(Index targetIndex, Category category) {
         this.targetIndex = targetIndex;
+        this.category = category;
     }
 
     @Override
     public CommandResult execute(Model model, ActiveAccount activeAccount) throws CommandException {
-        /*
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+        ObservableList<Expense> expenseList = activeAccount.getFilteredExpenseList();
+        ObservableList<Revenue> revenueList = activeAccount.getFilteredRevenueList();
```

Comment:
Try not to get the lists itself out as there are delete methods available in the ActiveAccount class, for example deleteExpense and deleteRevenue